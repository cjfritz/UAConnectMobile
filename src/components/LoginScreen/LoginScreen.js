import React, { Component } from 'react';
import { View, Image } from 'react-native';
import { connect } from 'react-redux';
import {
  Container,
  Form,
  Item,
  Input,
  Label,
  Button,
  H1,
} from 'native-base';
import { emailChanged, passwordChanged, loginUser } from '../../actions';
import styles from './LoginScreen.style';
import { LoadingModal } from '../common/LoadingModal/LoadingModal';
import imageLogo from '../../assets/images/ua_logo.png';

export class LoginScreen extends Component {
  render() {
    const {
      email,
      password,
      onEmailChange,
      onPasswordChange,
      loading,
      onLoginUser,
    } = this.props;

    return (
      <Container style={ styles.container }>
        <LoadingModal visible={ loading } loadingLabel='Logging in...' />
        <Image style={ styles.logo } source={ imageLogo } resizeMode='contain' />
        <View style={ { backgroundColor: 'transparent' } }>
          <Form>
            <Item style={ styles.item } floatingLabel>
              <Label style={ { color: 'white' } }>Email</Label>
              <Input
                keyboardType='email-address'
                style={ styles.input }
                onChangeText={ text => onEmailChange(text) }
                value={ email }
              />
            </Item>
          </Form>

          <Form>
            <Item style={ styles.item } floatingLabel>
              <Label style={ { color: 'white' } }>Password</Label>
              <Input
                style={ styles.input }
                secureTextEntry
                onChangeText={ text => onPasswordChange(text) }
                value={ password }
              />
            </Item>
          </Form>
          <Button
            style={ styles.loginButton }
            block
            onPress={ () => onLoginUser(email, password) }
          >
            <H1 style={ styles.text }>Sign In</H1>
          </Button>
        </View>
      </Container>
    );
  }
}

const mapDispatchToProps = dispatch => (
  ({
    onEmailChange: text => dispatch(emailChanged(text)),
    onPasswordChange: text => dispatch(passwordChanged(text)),
    onLoginUser: (email, password) => dispatch(loginUser({ email, password })),
  })
);

const mapStateToProps = state => {
  const {
    email,
    password,
    loading,
  } = state.auth;

  return {
    email,
    password,
    loading,
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(LoginScreen);
