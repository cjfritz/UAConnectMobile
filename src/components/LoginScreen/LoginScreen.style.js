import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  container: {
    alignContent: 'center',
    backgroundColor: '#67141A',
    flex: 1,
  },
  input: {
    color: 'white',
  },
  item: {
    marginLeft: '5%',
    marginRight: '5%',
  },
  loginButton: {
    alignItems: 'center',
    backgroundColor: 'white',
    justifyContent: 'center',
    marginLeft: '5%',
    marginRight: '5%',
    marginTop: '5%',
  },
  logo: {
    alignSelf: 'center',
    flex: 0.4,
    marginTop: '20%',
  },
  text: {
    color: 'black',
    fontSize: 20,
  },
});
