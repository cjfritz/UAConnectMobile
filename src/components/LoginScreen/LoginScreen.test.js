import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { LoginScreen } from './LoginScreen';

configure({ adapter: new Adapter() });

describe('<LoginScreen />', () => {
  it('renders correctly initially', () => {
    const renderer = shallow(<LoginScreen />);
    expect(renderer).toMatchSnapshot();
  });

  // it('renders with an email', () => {
  //   const props = {
  //   };
  // });
});
