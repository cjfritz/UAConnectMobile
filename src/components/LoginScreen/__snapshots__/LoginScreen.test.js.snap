// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginScreen /> renders correctly initially 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoginScreen />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <LoadingModal
          loadingLabel="Logging in..."
        />,
        <Component
          resizeMode="contain"
          source={
            Object {
              "testUri": "../../../src/assets/images/ua_logo.png",
            }
          }
          style={
            Object {
              "alignSelf": "center",
              "flex": 0.4,
              "marginTop": "20%",
            }
          }
        />,
        <Component
          style={
            Object {
              "backgroundColor": "transparent",
            }
          }
        >
          <Styled(Form)>
            <Styled(Item)
              floatingLabel={true}
              style={
                Object {
                  "marginLeft": "5%",
                  "marginRight": "5%",
                }
              }
            >
              <Styled(Label)
                style={
                  Object {
                    "color": "white",
                  }
                }
              >
                Email
              </Styled(Label)>
              <Styled(Input)
                onChangeText={[Function]}
                style={
                  Object {
                    "color": "white",
                  }
                }
              />
            </Styled(Item)>
          </Styled(Form)>
          <Styled(Form)>
            <Styled(Item)
              floatingLabel={true}
              style={
                Object {
                  "marginLeft": "5%",
                  "marginRight": "5%",
                }
              }
            >
              <Styled(Label)
                style={
                  Object {
                    "color": "white",
                  }
                }
              >
                Password
              </Styled(Label)>
              <Styled(Input)
                onChangeText={[Function]}
                secureTextEntry={true}
                style={
                  Object {
                    "color": "white",
                  }
                }
              />
            </Styled(Item)>
          </Styled(Form)>
          <Styled(Button)
            block={true}
            onPress={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "justifyContent": "center",
                "marginLeft": "5%",
                "marginRight": "5%",
                "marginTop": "5%",
              }
            }
          >
            <Styled(H1)
              style={
                Object {
                  "color": "black",
                  "fontSize": 20,
                }
              }
            >
              Sign In
            </Styled(H1)>
          </Styled(Button)>
        </Component>,
      ],
      "style": Object {
        "alignContent": "center",
        "backgroundColor": "#67141A",
        "flex": 1,
      },
      "virtual": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "loadingLabel": "Logging in...",
          "visible": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "resizeMode": "contain",
          "source": Object {
            "testUri": "../../../src/assets/images/ua_logo.png",
          },
          "style": Object {
            "alignSelf": "center",
            "flex": 0.4,
            "marginTop": "20%",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Styled(Form)>
              <Styled(Item)
                floatingLabel={true}
                style={
                  Object {
                    "marginLeft": "5%",
                    "marginRight": "5%",
                  }
                }
              >
                <Styled(Label)
                  style={
                    Object {
                      "color": "white",
                    }
                  }
                >
                  Email
                </Styled(Label)>
                <Styled(Input)
                  onChangeText={[Function]}
                  style={
                    Object {
                      "color": "white",
                    }
                  }
                />
              </Styled(Item)>
            </Styled(Form)>,
            <Styled(Form)>
              <Styled(Item)
                floatingLabel={true}
                style={
                  Object {
                    "marginLeft": "5%",
                    "marginRight": "5%",
                  }
                }
              >
                <Styled(Label)
                  style={
                    Object {
                      "color": "white",
                    }
                  }
                >
                  Password
                </Styled(Label)>
                <Styled(Input)
                  onChangeText={[Function]}
                  secureTextEntry={true}
                  style={
                    Object {
                      "color": "white",
                    }
                  }
                />
              </Styled(Item)>
            </Styled(Form)>,
            <Styled(Button)
              block={true}
              onPress={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "white",
                  "justifyContent": "center",
                  "marginLeft": "5%",
                  "marginRight": "5%",
                  "marginTop": "5%",
                }
              }
            >
              <Styled(H1)
                style={
                  Object {
                    "color": "black",
                    "fontSize": 20,
                  }
                }
              >
                Sign In
              </Styled(H1)>
            </Styled(Button)>,
          ],
          "style": Object {
            "backgroundColor": "transparent",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Styled(Item)
                floatingLabel={true}
                style={
                  Object {
                    "marginLeft": "5%",
                    "marginRight": "5%",
                  }
                }
              >
                <Styled(Label)
                  style={
                    Object {
                      "color": "white",
                    }
                  }
                >
                  Email
                </Styled(Label)>
                <Styled(Input)
                  onChangeText={[Function]}
                  style={
                    Object {
                      "color": "white",
                    }
                  }
                />
              </Styled(Item)>,
              "virtual": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Styled(Label)
                    style={
                      Object {
                        "color": "white",
                      }
                    }
                  >
                    Email
                  </Styled(Label)>,
                  <Styled(Input)
                    onChangeText={[Function]}
                    style={
                      Object {
                        "color": "white",
                      }
                    }
                  />,
                ],
                "floatingLabel": true,
                "style": Object {
                  "marginLeft": "5%",
                  "marginRight": "5%",
                },
                "virtual": undefined,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Email",
                    "style": Object {
                      "color": "white",
                    },
                    "virtual": undefined,
                  },
                  "ref": null,
                  "rendered": "Email",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "onChangeText": [Function],
                    "style": Object {
                      "color": "white",
                    },
                    "value": undefined,
                    "virtual": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Styled(Item)
                floatingLabel={true}
                style={
                  Object {
                    "marginLeft": "5%",
                    "marginRight": "5%",
                  }
                }
              >
                <Styled(Label)
                  style={
                    Object {
                      "color": "white",
                    }
                  }
                >
                  Password
                </Styled(Label)>
                <Styled(Input)
                  onChangeText={[Function]}
                  secureTextEntry={true}
                  style={
                    Object {
                      "color": "white",
                    }
                  }
                />
              </Styled(Item)>,
              "virtual": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Styled(Label)
                    style={
                      Object {
                        "color": "white",
                      }
                    }
                  >
                    Password
                  </Styled(Label)>,
                  <Styled(Input)
                    onChangeText={[Function]}
                    secureTextEntry={true}
                    style={
                      Object {
                        "color": "white",
                      }
                    }
                  />,
                ],
                "floatingLabel": true,
                "style": Object {
                  "marginLeft": "5%",
                  "marginRight": "5%",
                },
                "virtual": undefined,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Password",
                    "style": Object {
                      "color": "white",
                    },
                    "virtual": undefined,
                  },
                  "ref": null,
                  "rendered": "Password",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "onChangeText": [Function],
                    "secureTextEntry": true,
                    "style": Object {
                      "color": "white",
                    },
                    "value": undefined,
                    "virtual": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "block": true,
              "children": <Styled(H1)
                style={
                  Object {
                    "color": "black",
                    "fontSize": 20,
                  }
                }
              >
                Sign In
              </Styled(H1)>,
              "onPress": [Function],
              "style": Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "justifyContent": "center",
                "marginLeft": "5%",
                "marginRight": "5%",
                "marginTop": "5%",
              },
              "virtual": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Sign In",
                "style": Object {
                  "color": "black",
                  "fontSize": 20,
                },
                "virtual": undefined,
              },
              "ref": null,
              "rendered": "Sign In",
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <LoadingModal
            loadingLabel="Logging in..."
          />,
          <Component
            resizeMode="contain"
            source={
              Object {
                "testUri": "../../../src/assets/images/ua_logo.png",
              }
            }
            style={
              Object {
                "alignSelf": "center",
                "flex": 0.4,
                "marginTop": "20%",
              }
            }
          />,
          <Component
            style={
              Object {
                "backgroundColor": "transparent",
              }
            }
          >
            <Styled(Form)>
              <Styled(Item)
                floatingLabel={true}
                style={
                  Object {
                    "marginLeft": "5%",
                    "marginRight": "5%",
                  }
                }
              >
                <Styled(Label)
                  style={
                    Object {
                      "color": "white",
                    }
                  }
                >
                  Email
                </Styled(Label)>
                <Styled(Input)
                  onChangeText={[Function]}
                  style={
                    Object {
                      "color": "white",
                    }
                  }
                />
              </Styled(Item)>
            </Styled(Form)>
            <Styled(Form)>
              <Styled(Item)
                floatingLabel={true}
                style={
                  Object {
                    "marginLeft": "5%",
                    "marginRight": "5%",
                  }
                }
              >
                <Styled(Label)
                  style={
                    Object {
                      "color": "white",
                    }
                  }
                >
                  Password
                </Styled(Label)>
                <Styled(Input)
                  onChangeText={[Function]}
                  secureTextEntry={true}
                  style={
                    Object {
                      "color": "white",
                    }
                  }
                />
              </Styled(Item)>
            </Styled(Form)>
            <Styled(Button)
              block={true}
              onPress={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "white",
                  "justifyContent": "center",
                  "marginLeft": "5%",
                  "marginRight": "5%",
                  "marginTop": "5%",
                }
              }
            >
              <Styled(H1)
                style={
                  Object {
                    "color": "black",
                    "fontSize": 20,
                  }
                }
              >
                Sign In
              </Styled(H1)>
            </Styled(Button)>
          </Component>,
        ],
        "style": Object {
          "alignContent": "center",
          "backgroundColor": "#67141A",
          "flex": 1,
        },
        "virtual": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "loadingLabel": "Logging in...",
            "visible": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "resizeMode": "contain",
            "source": Object {
              "testUri": "../../../src/assets/images/ua_logo.png",
            },
            "style": Object {
              "alignSelf": "center",
              "flex": 0.4,
              "marginTop": "20%",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Styled(Form)>
                <Styled(Item)
                  floatingLabel={true}
                  style={
                    Object {
                      "marginLeft": "5%",
                      "marginRight": "5%",
                    }
                  }
                >
                  <Styled(Label)
                    style={
                      Object {
                        "color": "white",
                      }
                    }
                  >
                    Email
                  </Styled(Label)>
                  <Styled(Input)
                    onChangeText={[Function]}
                    style={
                      Object {
                        "color": "white",
                      }
                    }
                  />
                </Styled(Item)>
              </Styled(Form)>,
              <Styled(Form)>
                <Styled(Item)
                  floatingLabel={true}
                  style={
                    Object {
                      "marginLeft": "5%",
                      "marginRight": "5%",
                    }
                  }
                >
                  <Styled(Label)
                    style={
                      Object {
                        "color": "white",
                      }
                    }
                  >
                    Password
                  </Styled(Label)>
                  <Styled(Input)
                    onChangeText={[Function]}
                    secureTextEntry={true}
                    style={
                      Object {
                        "color": "white",
                      }
                    }
                  />
                </Styled(Item)>
              </Styled(Form)>,
              <Styled(Button)
                block={true}
                onPress={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "white",
                    "justifyContent": "center",
                    "marginLeft": "5%",
                    "marginRight": "5%",
                    "marginTop": "5%",
                  }
                }
              >
                <Styled(H1)
                  style={
                    Object {
                      "color": "black",
                      "fontSize": 20,
                    }
                  }
                >
                  Sign In
                </Styled(H1)>
              </Styled(Button)>,
            ],
            "style": Object {
              "backgroundColor": "transparent",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Styled(Item)
                  floatingLabel={true}
                  style={
                    Object {
                      "marginLeft": "5%",
                      "marginRight": "5%",
                    }
                  }
                >
                  <Styled(Label)
                    style={
                      Object {
                        "color": "white",
                      }
                    }
                  >
                    Email
                  </Styled(Label)>
                  <Styled(Input)
                    onChangeText={[Function]}
                    style={
                      Object {
                        "color": "white",
                      }
                    }
                  />
                </Styled(Item)>,
                "virtual": undefined,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <Styled(Label)
                      style={
                        Object {
                          "color": "white",
                        }
                      }
                    >
                      Email
                    </Styled(Label)>,
                    <Styled(Input)
                      onChangeText={[Function]}
                      style={
                        Object {
                          "color": "white",
                        }
                      }
                    />,
                  ],
                  "floatingLabel": true,
                  "style": Object {
                    "marginLeft": "5%",
                    "marginRight": "5%",
                  },
                  "virtual": undefined,
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Email",
                      "style": Object {
                        "color": "white",
                      },
                      "virtual": undefined,
                    },
                    "ref": null,
                    "rendered": "Email",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "onChangeText": [Function],
                      "style": Object {
                        "color": "white",
                      },
                      "value": undefined,
                      "virtual": undefined,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Styled(Item)
                  floatingLabel={true}
                  style={
                    Object {
                      "marginLeft": "5%",
                      "marginRight": "5%",
                    }
                  }
                >
                  <Styled(Label)
                    style={
                      Object {
                        "color": "white",
                      }
                    }
                  >
                    Password
                  </Styled(Label)>
                  <Styled(Input)
                    onChangeText={[Function]}
                    secureTextEntry={true}
                    style={
                      Object {
                        "color": "white",
                      }
                    }
                  />
                </Styled(Item)>,
                "virtual": undefined,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <Styled(Label)
                      style={
                        Object {
                          "color": "white",
                        }
                      }
                    >
                      Password
                    </Styled(Label)>,
                    <Styled(Input)
                      onChangeText={[Function]}
                      secureTextEntry={true}
                      style={
                        Object {
                          "color": "white",
                        }
                      }
                    />,
                  ],
                  "floatingLabel": true,
                  "style": Object {
                    "marginLeft": "5%",
                    "marginRight": "5%",
                  },
                  "virtual": undefined,
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Password",
                      "style": Object {
                        "color": "white",
                      },
                      "virtual": undefined,
                    },
                    "ref": null,
                    "rendered": "Password",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "onChangeText": [Function],
                      "secureTextEntry": true,
                      "style": Object {
                        "color": "white",
                      },
                      "value": undefined,
                      "virtual": undefined,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "block": true,
                "children": <Styled(H1)
                  style={
                    Object {
                      "color": "black",
                      "fontSize": 20,
                    }
                  }
                >
                  Sign In
                </Styled(H1)>,
                "onPress": [Function],
                "style": Object {
                  "alignItems": "center",
                  "backgroundColor": "white",
                  "justifyContent": "center",
                  "marginLeft": "5%",
                  "marginRight": "5%",
                  "marginTop": "5%",
                },
                "virtual": undefined,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Sign In",
                  "style": Object {
                    "color": "black",
                    "fontSize": 20,
                  },
                  "virtual": undefined,
                },
                "ref": null,
                "rendered": "Sign In",
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
